gclient/src/api/listener/mod.rs

```diff
@@ -58,30 +62,64 @@ impl From<GenDispatchStatus> for DispatchStatus {
 }
 
 impl DispatchStatus {
+    /// Check whether `DispatchStatus` is
+    /// [`Success`](DispatchStatus::Success)ful.
     pub fn succeed(&self) -> bool {
         matches!(self, DispatchStatus::Success)
     }
 
+    /// Check whether `DispatchStatus` is [`Failed`](DispatchStatus::Failed).
     pub fn failed(&self) -> bool {
         matches!(self, DispatchStatus::Failed)
     }
 
+    /// Check whether `DispatchStatus` is
+    /// [`NotExecuted`](DispatchStatus::NotExecuted).
     pub fn not_executed(&self) -> bool {
         matches!(self, DispatchStatus::NotExecuted)
     }
 }
 
+/// Events processing trait.
+///
+/// This trait provides default implementations for processing several events
+/// propagated through the network.
+///
+/// See implementation example in [`EventListener`].
 #[async_trait(?Send)]
 pub trait EventProcessor {
+    /// This function is called if a received event has an unexpected type.
+    ///
+    /// # Examples
+    ///
+    /// Implement this function to return a not-found error:
+    ///
+    /// ```
+    /// # use gclient::Error;
+    /// fn not_waited() -> Error {
+    ///     Error::EventNotFoundInIterator
+    /// }
+    /// ```
     fn not_waited() -> Error;
 
+    /// Event processing function.
```

Comment:
More explanation please
Like it applies on each event, returns result to be catched etc

Reply:
Added